File: src/frankversnel/processing/collision/component/Collidable.java
package frankversnel.processing.collision.component;

import frankversnel.processing.GameObject;
import frankversnel.processing.component.Component;

public abstract class Collidable extends Component {

	public Collidable(GameObject gameObject) {
		super(gameObject);
	}

	public abstract boolean collidesWith(Collidable other);

}

File: src/main/scala/org/frankversnel/poortjes/collision/collidable.scala
package org.frankversnel.poortjes.collision

import org.newdawn.slick.geom.Shape
import org.slf4j.scala.Logging;

import org.frankversnel.poortjes._

trait Collidable extends Transform with Dimension with Logging {

	def collidesWith(otherCollidable: Collidable): Boolean = {
		boundingBox.intersects(otherCollidable.boundingBox)
	}

	def onCollision(collider: Collidable, update: Update): Unit

	protected def boundingBox: Shape
}
